Tests/Services/NfieldSurveyInterviewSimulationTests.cs

```diff
@@ -60,5 +69,302 @@ public async Task TestGetAsync()
+            var content = new StringContent(JsonConvert.SerializeObject(activityStatus));
+
+            _mockedHttpClient.Setup(client =>
+                client.PostAsync(new Uri(ServiceAddress, $"surveys/{_surveyId}/InterviewSimulations/StartInterviewSimulations"), It.IsAny<MultipartFormDataContent>()))
+                .Returns(CreateTask(HttpStatusCode.OK, content));
+
+            _mockedHttpClient
+                .Setup(client => client.GetAsync(It.IsAny<Uri>()))
+                .Returns(Task.Factory.StartNew(
+                    () =>
+                    new HttpResponseMessage(HttpStatusCode.OK)
+                    {
+                        Content = new StringContent(JsonConvert.SerializeObject(new
+                        {
+                            ActivityId = "activityId",
+                            Status = 2
+                        }))
+                    })).Verifiable();
+
+            var result = await _target.StartSimulationAsync(_surveyId, new InterviewSimulationFiles
+            {
+                InterviewsCount = 1,
+                UseOriginalSample = true,
+                EnableReporting = true,
+                HintsFilePath = HintsFilePath,
+                SampleDataFilePath = SampleDataFilePath
+            });
+
+            _mockedHttpClient.Verify(client => client.PostAsync(It.IsAny<Uri>(), It.Is<MultipartFormDataContent>(c =>
+                c.IsMimeMultipartContent() &&
+                c.Count() == 5 &&
+                IsPropertyAvailable(c, new ContentProperty { Name = "InterviewsCount", NameValue = "1" }) &&
+                IsPropertyAvailable(c, new ContentProperty { Name = "UseOriginalSample", NameValue = "true" }) &&
+                IsPropertyAvailable(c, new ContentProperty { Name = "EnableReporting", NameValue = "true" }) &&
+                IsFilePropertyAvailable(c, new ContentProperty { Name = "HintsFile", NameValue = "hintsFile", FileName = HintsFileName, FileContent = HintsData }) &&
+                IsFilePropertyAvailable(c, new ContentProperty { Name = "SampleDataFile", NameValue = "sampleDataFile", FileName = SampleDataFileName, FileContent = SampleData })
+                )));
+        }
+
+        #endregion
+
+        private static bool IsPropertyAvailable(MultipartFormDataContent content, ContentProperty contentProperty)
+        {
+            foreach (var contentItem in content)
+            {
+                var contentHeader = contentItem.Headers.ContentDisposition.Parameters;
+                if (contentHeader.FirstOrDefault(c => c.Name == "name" && c.Value == contentProperty.Name) != null)
+                {
+                    var contentString = contentItem.ReadAsStringAsync().Result;
+                    if (string.Compare(contentString, contentProperty.NameValue, StringComparison.OrdinalIgnoreCase) != 0)
+                        return false;
+
+                    return true;
```

Comment:
This could be simplified by just returning `return string.Compare(...) == 0;`

Reply:
:thumbsup: